### PR TITLE
ImageMagick TGA support fix

### PR DESCRIPTION
**Pull request by rmottola**
seek for trailing signature and recognize v2 TGA files and hint that to ImageMagick

works for v2 TGA files

### CODE REVIEW

**Comment by rmottola** [comment]
@eukara I think this might interest you

File: Source/GSImageMagickImageRep.m [hunk 1]
```diff
@@ -139,6 +142,14 @@ + (NSArray*) imageRepsWithData: (NSData *)data allImages: (BOOL)allImages
   QueryColorDatabase("none", &imageinfo->background_color, exception);
 #endif
 
+  bzero(signature, 32);
+  [data getBytes:signature range:NSMakeRange([data length]-18, 18)];
```
Comment (fredkiefer) [inline Source/GSImageMagickImageRep.m:146]: Here some additional spaces would improve readability

File: Source/GSImageMagickImageRep.m [hunk 2]
```diff
@@ -129,7 +131,9 @@ + (NSArray*) imageRepsWithData: (NSData *)data allImages: (BOOL)allImages
 
   ExceptionInfo *exception = AcquireExceptionInfo();
   ImageInfo *imageinfo = CloneImageInfo(NULL);
-  Image *images, *image;
+  Image *images;
+  Image *image;
+  char signature[32];
```
Comment (fredkiefer) [inline Source/GSImageMagickImageRep.m:136]: Why is the signature 32 bytes long when you only use 18 of them?

File: Source/GSImageMagickImageRep.m [hunk 3]
```diff
@@ -139,6 +143,14 @@ + (NSArray*) imageRepsWithData: (NSData *)data allImages: (BOOL)allImages
   QueryColorDatabase("none", &imageinfo->background_color, exception);
 #endif
 
+  memset(signature, 0, 32);
+  [data getBytes: signature range: NSMakeRange([data length] - 18, 18)];
+  if (strncmp(signature, "TRUEVISION-XFILE.", 17) == 0)
+    {
+      NSWarnLog(@"Targa file detected!, giving a magick hint...");
+      strcpy(imageinfo->magick, "TGA");
```
Comment (fredkiefer) [inline Source/GSImageMagickImageRep.m:151]: Although it is pretty save here, we normally don't use `strcpy`. Rather something like `strncpy`. Here it makes no difference but static code analysis tools may complain.

**Review by fredkiefer — APPROVED** [review]

File: Source/GSImageMagickImageRep.m (hunk 1 above)
Comment (rmottola) [reply Source/GSImageMagickImageRep.m]: Spaced better!

File: Source/GSImageMagickImageRep.m (hunk 2 above)
Comment (rmottola) [reply Source/GSImageMagickImageRep.m:136]: As a safety for future uses, we could try to handle more than one signature.
I was looking and there are other files misbehaving, so one could reuse it.

If you wish I could reduce it to exactly 18, I don't think 14bytes are a big issues though.

File: Source/GSImageMagickImageRep.m (hunk 3 above)
Comment (rmottola) [reply Source/GSImageMagickImageRep.m:151]: > Although it is pretty save here, we normally don't use `strcpy`. Rather something like `strncpy`. Here it makes no difference but static code analysis tools may complain.

is it needed even if the right value is a constant? looks overkill, a code analyzer shouldn't complain.

**Comment by eukara** [comment]
Thank you very much Riccardo!

File: Source/GSImageMagickImageRep.m (hunk 2 above)
Comment (rmottola) [reply Source/GSImageMagickImageRep.m]: @fredkiefer to appease both, I defined a local constant. So I declare for now just the 18 bytes needed for TGA, but it is more evident that if you tinker with it, you need to update it. And memset will be correct too. Fine?

**Comment by rmottola** [comment]
Tested on ImageMagick 6 & 7. Going to merge. Details can be smoothed out, @fredkiefer!

**Pull request merged**